<a href="https://colab.research.google.com/github/sinku29/Data-Science-Assignment/blob/main/Hypo_BuyerRatio.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [1]:
import pandas as pd
import numpy as np
from scipy import stats
from scipy.stats import norm

In [11]:
data=pd.read_csv("BuyerRatio.csv")
print(data)

  Observed Values  East  West  North  South
0           Males    50   142    131     70
1         Females   435  1523   1356    750


In [15]:
table = data.iloc[:,1:6]
print(table)

   East  West  North  South
0    50   142    131     70
1   435  1523   1356    750


In [28]:
Chisquares_results=stats.chi2_contingency(table)
Chisquares_results

Chi2ContingencyResult(statistic=1.595945538661058, pvalue=0.6603094907091882, dof=3, expected_freq=array([[  42.76531299,  146.81287862,  131.11756787,   72.30424052],
       [ 442.23468701, 1518.18712138, 1355.88243213,  747.69575948]]))

In [29]:
print("Chi-square statistic:", Chisquares_results[0])
print("P-value:", Chisquares_results[1])
print("Degrees of freedom:", Chisquares_results[2])
print("Expected frequencies:")
print(Chisquares_results[3])

Chi-square statistic: 1.595945538661058
P-value: 0.6603094907091882
Degrees of freedom: 3
Expected frequencies:
[[  42.76531299  146.81287862  131.11756787   72.30424052]
 [ 442.23468701 1518.18712138 1355.88243213  747.69575948]]


p value is 0.6603  Since the p-value is greater than the common significance level of 0.05 , we fail to reject the null hypothesis.

Since we fail to reject the null hypothesis, we conclude that there is not enough evidence to suggest that the male-female buyer ratios significantly differ across regions. In other words, based on the data, it appears that the male-female buyer ratios are similar across the four different regions.